jobrunner/cli/local_run.py

```diff
@@ -406,14 +406,31 @@ def create_job_request_and_jobs(project_dir, actions, force_run_dependencies):
     # changes below then consider what, if any, the appropriate corresponding
     # changes might be for production jobs.
     project = parse_and_validate_project_file(project_file_path.read_bytes())
-    current_jobs = get_latest_job_for_each_action(job_request.workspace)
-    new_jobs = get_new_jobs_to_run(job_request, project, current_jobs)
-    assert_new_jobs_created(new_jobs, current_jobs)
+    latest_jobs = get_latest_job_for_each_action(job_request.workspace)
+
+    # On the server out-of-band deletion of an existing output is considered an error, so we ignore that case when
+    # scheduling and allow jobs with missing dependencies to fail loudly when they are actually run. However for local
+    # running we should allow researchers to delete outputs on disk and automatically rerun the actions that create
+    # if they are needed. So here we check whether any files are missing for completed actions and, if so, treat them
+    # as though they had not been run -- this will automatically trigger a rerun.
+    latest_jobs_with_files_present = filter(
+        all_output_files_present(project_dir), latest_jobs
+    )
```

Comment:
I don't think this will work because `filter()` gives you an iterator not a list and it's being passed to two functions below, so the first will consume it. Wrapping it in a `list()` call will do the trick, though you might decide that with the extra nesting the end result is less clear than a list comprehension would have been.

Reply:
Ugh. Well spotted -- thank you. The UX for things-you-can-iterate-over in Python is really shot to pieces. `yield` is amazing when it's just what you want, but the knock-on implications like this really make me doubt that it's a good trade-off.